Source/Core/DolphinQt/Debugger/CodeDiffDialog.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDialog>
+
+#include "Common/CommonTypes.h"
+#include "Core/PowerPC/Profiler.h"
+
+class CodeWidget;
+class QCheckBox;
+class QLabel;
+class QListWidget;
+
+struct Diff
+{
+  u32 addr;
+  std::string symbol;
+  u32 hits;
+};
+
+class CodeDiffDialog : public QDialog
+{
+  Q_OBJECT
+
+  struct _AddrOp

```

Comment:
Is there any reason not to make this a function?

Reply:
I'm not familiar with C++ optimizations

I'll be doing some googling.

@TryTwo was this done for a performance reason?

Reply:
I opt to leave this as is.
It was done to simplify having a predicate when used here:

`auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);`
Otherwise we would explicitly have to pass in the iter and the list where AddrOP is. 
Instead as is, its overwriting the operator.

Reply:
Yes, that was the point. Defining a function `AddrOp` such as:
```c++
static bool AddrOp(const Diff& iter, const std::string& val) {
    return iter.symbol < val;
}
```

It should work exactly the same without having to create a new structure:
```c++
auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, CodeDiffDialog::AddrOP);
```

If that `AddrOp` helper is only used in `lower_bound` I'd recommend to implement the `operator<` instead in the `struct Diff`. That way, you'll just have to write:
```c++
auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol);
```

Reply:
Getting LNK1169 and LNK2005
```
Error	LNK2005	"bool __cdecl operator<(struct Diff const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??M@YA_NAEBUDiff@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) already defined in CodeDiffDialog.obj	Dolphin	X:\dev\dolphin\Source\Core\DolphinQt\CodeWidget.obj	1	
```

Completely lost as to why.
The operator is defined in `CodeDiffDialog.h` outside and above the Class definition, just below struct Diff.
The only AddrOp struct has been deleted.
I removed AddrOp references from every other place in the code.
Clean solution and rebuild and getting this.

Is this something caused by MocQt I need to update? I looked through those files but nothing stuck out.

Reply:
The error means that the operator is defined more than once (apparently in CodeWidget and CodeDiffDialog) and I don't think it's related to Qt.

I would have put the operator function inside the `Diff` struct.